packages/server/graphql/types/AzureDevOpsIntegration.ts

```diff
@@ -0,0 +1,158 @@
+      type: TeamMemberIntegrationAuthOAuth2,
+      resolve: async (
+        {teamId, userId}: {teamId: string; userId: string},
+        _args: unknown,
+        {dataLoader}
+      ) => {
+        return dataLoader
+          .get('teamMemberIntegrationAuths')
+          .load({service: 'azureDevOps', teamId, userId})
+      }
+    },
+    id: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'Composite key in ado:teamId:userId format',
+      resolve: ({teamId, userId}: {teamId: string; userId: string}) => `ado:${teamId}:${userId}`
+    },
+    isActive: {
+      description: 'true if the auth is valid, else false',
+      type: new GraphQLNonNull(GraphQLBoolean),
+      resolve: ({accessToken}) => !!accessToken
+    },
+    accessToken: {
+      description:
+        'The access token to Azure DevOps. null if no access token available or the viewer is not the user',
+      type: GraphQLID
+      // Add resolver
+    },
+    accountId: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'The Azure DevOps account ID'
+    },
+    instanceIds: {
+      type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(GraphQLID))),
+      description: 'The Azure DevOps instance IDs that the user has granted'
+    },
+    createdAt: {
+      type: new GraphQLNonNull(GraphQLISO8601Type),
+      description: 'The timestamp the provider was created'
+    },
+    teamId: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'The team that the token is linked to'
+    },
+    updatedAt: {
+      type: new GraphQLNonNull(GraphQLISO8601Type),
+      description: 'The timestamp the token was updated at'
+    },
+    userId: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'The user that the access token is attached to'
+    },
+    workItems: {
+      type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(AzureDevOpsWorkItem))),
+      description:
+        'A list of work items coming straight from the azure dev ops integration for a specific team member',
+      resolve: async (
+        {teamId, userId}: AzureDevOpsAuth,
+        {},
+        {authToken, dataLoader}: GQLContext
+      ) => {
+        const viewerId = getUserId(authToken)
+        if (viewerId !== userId) {
+          const err = new Error('Cannot access another team members issues')
+          standardError(err, {tags: {teamId, userId}, userId: viewerId})
+          return connectionFromTasks([], 0, err)
+        }
+        const workItems = await dataLoader.get('azureDevOpsAllWorkItems').load({teamId, userId})
+        if (workItems === undefined) {
+          return []
+        } else {
+          const userStories = Array.from(
+            workItems.map((workItem) => {
+              return {
+                id: workItem.id.toString(),
+                url: workItem.url
+              }
+            })
+          )
+          return userStories
+        }
+      }
+    },
+    cloudProvider: {
```

Comment:
-2 I think you would not have a cloudProvider, would you? This is the one global instance which is publicly available. An example would be gitlab.com as the `cloudProvider` vs. the self hosted GitLab instances as `sharedProvider`. As far as I understand all Azure DevOps instances would be `sharedProvider`s since they differ by the tenant. There is no single "azure.microsoft.com" instance where everyone can authenticate against, is there?

Reply:
Agree completely. Thank you for your clear explanation on the difference between `cloudProvider` and `sharedProvider`. Change is made.